pkg/segment/writer/segwriter.go

```diff
@@ -356,6 +356,23 @@ func (ss *SegStore) doLogEventFilling(ple *ParsedLogEvent, tsKey *string) (bool,
 				ss.checkAddDictEnc(colWip, colWip.cbuf[startIdx:colWip.cbufidx], ss.wipBlock.blockSummary.RecCount, startIdx, false)
 			}
 			ss.updateColValueSizeInAllSeenColumns(cname, colWip.cbufidx-startIdx)
+		case VALTYPE_ENC_BOOL[0]:
+			startIdx := colWip.cbufidx
+			copy(colWip.cbuf[startIdx:], ple.allCvalsTypeLen[i][0:1])
+			colWip.cbufidx += 1
+			copy(colWip.cbuf[colWip.cbufidx:], ple.allCvals[i][0:1])
+			colWip.cbufidx += 1
+			boolVal := utils.BytesToBoolLittleEndian(ple.allCvals[i][0:1])
+			var asciiBytesBuf bytes.Buffer
+			_, err := fmt.Fprintf(&asciiBytesBuf, "%v", boolVal)
+			if err != nil {
+				return false, utils.TeeErrorf("doLogEventFilling: cannot write boolVal %v: %v", boolVal, err)
+			}
+			addSegStatsBool(segstats, cname, asciiBytesBuf.Bytes())
```

Comment:
Why do you have to do so much work here to get `asciiBytesBuf.Bytes()`? Isn't it always going to be 1?

Reply:
bool value can be true or false , we need to get the value not just the length